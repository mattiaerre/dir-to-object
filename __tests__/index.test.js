const { join } = require('path');
const dirToObject = require('../src');

describe('dirToObject', () => {
  const scenarios = [
    {
      config: { dirPath: join(__dirname, '../__mocks__/foo') },
      description: 'config w/ dirPath only'
    },
    {
      config: {
        canAdd: data => data.resolve && data.type,
        dirPath: join(__dirname, '../__mocks__/foo')
      },
      description: 'config w/ canAdd and dirPath'
    }
  ];

  scenarios.forEach(({ config, description }) => {
    it(description, () => {
      expect(dirToObject(config)).toMatchSnapshot();
    });
  });
});
