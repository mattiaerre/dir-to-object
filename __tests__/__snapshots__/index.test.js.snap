// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`dirToObject to match snapshot 1`] = `
Object {
  "info": Object {
    "name": "dir-to-object",
    "version": "1.0.0-alpha.3",
  },
}
`;
